### PR TITLE
Handle ASCII-encoded segmentation UPIDs correctly

### PR DESCRIPTION
**Pull request by blahspam**

### CODE REVIEW

**Comment by blahspam** [comment]
@sbowman Can you do a quick review of my changes... and sign the CLA so I can eventually merge this?

I've modified your approach and changed it such that the models always use UTF8 but converts from Latin1 on decode and back on encode. This resolves a number of issues that your original fix didn't address around XML/JSON representation and lossless decoding and encoding.

Thanks.

File: pkg/scte35/segmentation_upid.go [hunk 1]
```diff
@@ -168,18 +170,15 @@ func (upid *SegmentationUPID) Name() string {
...
-	for _, b := range b {
-		var stringVal string
-		if utf8.Valid([]byte{b}) {
-			stringVal = string(b)
-		} else {
-			// non valid ascii byte element, it is represented as a dot
-			stringVal = "."
+	rs := make([]byte, 0, len(b))
+	for i := range b {
+		if b[i] <= 127 {
```
Comment (sbowman) [inline pkg/scte35/segmentation_upid.go:175]: I think this is supposed to be ASCII printable characters, which means the line should be something like:

    if b[i] > 31 && b[i] < 127

File: pkg/scte35/segmentation_upid.go [hunk 2]
```diff
@@ -168,18 +170,15 @@ func (upid *SegmentationUPID) Name() string {
...
+			rs[i] = b[i]
+			continue
 		}
-		stringsValues = append(stringsValues, stringVal)
+		rs[i] = byte(46) // "."
```
Comment (sbowman) [inline pkg/scte35/segmentation_upid.go:179]: You can simplify this:

    rs[i] = '.'

**Review by sbowman — COMMENTED** [review]
I think the ASCII is supposed to be printable characters, based on page 69 of SCTE_35_2022b.pdf

**Comment by blahspam** [comment]
Made changes per your feedback. If theres nothing else I'll merge it in tomorrow.

**Pull request merged**